src/app/services/user.service.ts

```diff
@@ -1,23 +1,42 @@
-import { Injectable } from '@angular/core';
-import {HttpClient} from "@angular/common/http";
+import {Injectable, signal, Signal, WritableSignal} from '@angular/core';
+import {HttpClient, HttpErrorResponse} from "@angular/common/http";
 import {BehaviorSubject, first, Observable} from "rxjs";
-import {UserModel} from "../shared/user.model";
+import {UserModel} from "../shared/models/user.model";
 import {environment} from "../../environments/enviornment";
+import {TaskModel} from "../shared/models/task.model";
+import {LoginCredentialsModel} from "../shared/models/login-credentials.model";
+import {AuthorizedUserModel} from "../shared/models/authorized-user.model";
 
 @Injectable({
   providedIn: 'root'
 })
 export class UserService {
   userSubject: BehaviorSubject<UserModel[]> = new BehaviorSubject<UserModel[]>([]);
   users$: Observable<UserModel[]> = this.userSubject.asObservable();
+  currentUserSig: WritableSignal<UserModel | undefined | null> = signal<UserModel | undefined | null>(undefined);
```

Comment:
why undefined and null, at most having either undefined or null should be sufficient, no?